network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/failure/ServerMigrationMonitor.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.failure
+
+import com.duckduckgo.common.utils.ConflatedJob
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.di.scopes.VpnScope
+import com.duckduckgo.mobile.android.vpn.service.VpnServiceCallbacks
+import com.duckduckgo.mobile.android.vpn.state.VpnStateMonitor.VpnStopReason
+import com.duckduckgo.networkprotection.api.NetworkProtectionState
+import com.duckduckgo.networkprotection.impl.configuration.WgTunnelConfig
+import com.duckduckgo.networkprotection.impl.configuration.WgVpnControllerService
+import com.duckduckgo.networkprotection.impl.configuration.asServerDetails
+import com.duckduckgo.networkprotection.impl.di.ProtectedVpnControllerService
+import com.duckduckgo.networkprotection.impl.pixels.NetworkProtectionPixels
+import com.squareup.anvil.annotations.ContributesMultibinding
+import javax.inject.Inject
+import kotlin.time.Duration.Companion.minutes
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import logcat.logcat
+
+@ContributesMultibinding(VpnScope::class)
+class ServerMigrationMonitor @Inject constructor(
+    @ProtectedVpnControllerService private val wgVpnControllerService: WgVpnControllerService,
+    private val wgTunnelConfig: WgTunnelConfig,
+    private val networkProtectionState: NetworkProtectionState,
+    private val dispatcherProvider: DispatcherProvider,
+    private val networkProtectionPixels: NetworkProtectionPixels,
+) : VpnServiceCallbacks {
+    private val job = ConflatedJob()
+    private var migrating = false
+
+    override fun onVpnStarted(coroutineScope: CoroutineScope) {
+        job += coroutineScope.launch(dispatcherProvider.io()) {
+            startMonitor()
+        }
+    }
+
+    override fun onVpnReconfigured(coroutineScope: CoroutineScope) {
+        job += coroutineScope.launch(dispatcherProvider.io()) {
+            if (migrating) {
```

Comment:
@aitorvs I’ve updated the way we emit pixels. 
If after restart() is called for migration:
- onVPNReconfigured is called -> it’s a success. We cleared the config, so we will reach this point if we manage to get a server from /server
- onVPNStartFailed is called -> it’s a failure. We cleared the config. If we are unable to get anything from /register, migration failed.